### PR TITLE
Added Kafka clients instrumentation blog post

### PR DESCRIPTION
**Pull request by ppatierno**
This PR addresses #1668 about writing a blog post on Kafka clients instrumentation with OpenTelemetry.

### CODE REVIEW

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (svrnm) [inline content/en/blog/2022/instrument-kafka-clients/index.md:8]: > the central piece in a distributed system.

the central piece for what? I as a reader have the feeling there is something missing or should it be "a central piece"?

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (svrnm) [inline content/en/blog/2022/instrument-kafka-clients/index.md:9]: communicate **with** each other.

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (svrnm) [inline content/en/blog/2022/instrument-kafka-clients/index.md:16]: the OpenTelemetry project -> OpenTelemetry

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (svrnm) [inline content/en/blog/2022/instrument-kafka-clients/index.md:17]: Although this it the otel blog, the reader might not know what semantic conventions are. Maybe skip that for now and rephrase that to:

This is where OpenTelemetry comes into the picture: it provides multiple instrumentation libraries for adding tracing to messaging based applications. Of course, there is one for Apache Kafka clients.

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (svrnm) [inline content/en/blog/2022/instrument-kafka-clients/index.md:20]: But usually, the architecture can even be more complicated, to But usually, the architecture can even be more complicated:

I have the feeling that the `:` makes it clearer to the reader that what is following is an explanation of the things said in the sentence before.

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (svrnm) [inline content/en/blog/2022/instrument-kafka-clients/index.md:21]: could be -> is

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (svrnm) [inline content/en/blog/2022/instrument-kafka-clients/index.md:22]: Why 'Even'? I think you can skip that.

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (svrnm) [inline content/en/blog/2022/instrument-kafka-clients/index.md:12]: strimzi is another cncf project, right?

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (svrnm) [inline content/en/blog/2022/instrument-kafka-clients/index.md:24]: we are going to see -> you will learn.

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (svrnm) [inline content/en/blog/2022/instrument-kafka-clients/index.md:98]: > This is really useful when using a tracing UI (i.e. Jaeger UI) for showing data.

The service name is kind of mandatory, maybe you can phrase this even stronger?

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (svrnm) [inline content/en/blog/2022/instrument-kafka-clients/index.md:121]: above you ask people to use the environment variable / system property, now you're setting the SERVICE_NAME via code as well, that might confuse the user.

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (svrnm) [inline content/en/blog/2022/instrument-kafka-clients/index.md:179]: I personally would like seeing that automatic way first, people (including myself) are lazy, so having something they can do within a few "clicks" is always a better hook than copying a bunch of code snippets.

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (svrnm) [inline content/en/blog/2022/instrument-kafka-clients/index.md:212]: isn't there some docker images people can use, or even a docker compose that build everything for them?

**Review by svrnm — CHANGES_REQUESTED** [review]
Thanks for writing this up! I provided some feedback via my review.

Please also run `npx prettier --write content/en/blog/2022/instrument-kafka-clients/index.md` to get some automatic style updates (line width, etc.)

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (ppatierno) [reply content/en/blog/2022/instrument-kafka-clients/index.md:8]: The idea behind this was to say that it's used a the "nervous" system for communication between applications in a distributed system.

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (ppatierno) [reply content/en/blog/2022/instrument-kafka-clients/index.md:12]: Yes it is. What's the point you would like to make? Maybe that there is an "integration" between Strimzi and OpenTelemetry as part of the CNCF? Adding something like this to the blog post?

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (ppatierno) [reply content/en/blog/2022/instrument-kafka-clients/index.md:98]: Actually in one of the previous discussion with other committers (I don't remember where and who :-D) they said that I should not make the point that service name is mandatory, which is kind of true, but at same time I agree with you. In general, it doesn't make sense not having a service name. For example, you could skip setting it but then in the Jaeger UI, for example, you have "unknown service" in the drop down list for selecting the service for getting the traces.

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (ppatierno) [reply content/en/blog/2022/instrument-kafka-clients/index.md:121]: This is the section related to not using the SDK auto-configuration (which relies on env vars as well) but it's about creating the `OpenTelemetry` instance programmatically from your own code.

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (ppatierno) [reply content/en/blog/2022/instrument-kafka-clients/index.md:179]: That's a good point. I will re-arrange the order!

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (ppatierno) [reply content/en/blog/2022/instrument-kafka-clients/index.md:212]: There are ... some of them use Kafka images from the Strimzi project, others use the Confluent one. I would like to be less opinionated and using directly the upstream project.

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (ppatierno) [reply content/en/blog/2022/instrument-kafka-clients/index.md]: I thought it was useful to mention the messaging semantic conventions because I am referring to them in the `Instrumentation in action` section as well [here](https://github.com/open-telemetry/opentelemetry.io/pull/1677/files#diff-1713a18cb9f06d7587ef469a52bb57100157ee894ee5a3aa6bf40ecd46147797R277), to explain that current "output" in terms of trace, spans and relationship between them is evolving and going to change with the OTEP.

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (ppatierno) [reply content/en/blog/2022/instrument-kafka-clients/index.md]: Done!

**Comment by ppatierno** [comment]
@svrnm I ran the "prettier" command but imho it makes review harder. As our experience about writing on the Strimzi blog, we prefer having "one sentence per line" because you can comment on each sentence directly for providing feedback.
Maybe my mistake was to run the "prettier" too soon after your request. I should have run it at the end, when the PR was finally approved.

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (svrnm) [reply content/en/blog/2022/instrument-kafka-clients/index.md]: I would be ok with using that term "nervous system", I guess most people understand that picture.

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (svrnm) [reply content/en/blog/2022/instrument-kafka-clients/index.md]: then put the instrumentation first and the semantic conventions second? People are more interested in the first I would say.

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (svrnm) [reply content/en/blog/2022/instrument-kafka-clients/index.md]: yeah, "kind of mandatory", maybe write something like "It is recommended to set it"?

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (svrnm) [reply content/en/blog/2022/instrument-kafka-clients/index.md:188]: ah, gotcha

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (svrnm) [reply content/en/blog/2022/instrument-kafka-clients/index.md]: What I wanted to say: is there one in your sample github repo or in one you create for this blog post, people can just copy that and run it (like we did it in the nginx blog where we give something people can follow step-by-step from 0)

**Comment by svrnm** [comment]
> @svrnm I ran the "prettier" command but imho it makes review harder. As our experience about writing on the Strimzi blog, we prefer having "one sentence per line" because you can comment on each sentence directly for providing feedback. Maybe my mistake was to run the "prettier" too soon after your request. I should have run it at the end, when the PR was finally approved.

Makes sense, then let's leave it like that and we run the prettier before merging.

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (ppatierno) [reply content/en/blog/2022/instrument-kafka-clients/index.md]: No, because most of my development with Kafka is always with running it locally without Docker compose (I don't have it installed at all :-)). The other is about using Kubernetes when I worked on Strimzi but using Kubernetes is really useless and too much for the purpose of the blog post.

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (svrnm) [reply content/en/blog/2022/instrument-kafka-clients/index.md]: first of all I wanted to verify that it's another OSS solution and not something commercial.

we might consider cross-posting this blog post in a second step on the cncf blog?

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (svrnm) [reply content/en/blog/2022/instrument-kafka-clients/index.md]: ok, then let's leave it like that!

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (ppatierno) [reply content/en/blog/2022/instrument-kafka-clients/index.md]: We can for sure cross-posting from our Strimzi official web site. We have never published a post on the CNCF blog.

**Comment by ppatierno** [comment]
@svrnm I think I addressed your comments, is there anything more? Any other maintainers to review this? At this point I should also change the publication date (which is tomorrow Sep 5th).

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (svrnm) [inline content/en/blog/2022/instrument-kafka-clients/index.md:3]: Replace with "Instrumenting Apache Kafka clients" to make the short title even shorter, we did the same for many other blog posts, see on the left side menu here:

https://deploy-preview-1677--opentelemetry.netlify.app/blog/2022/instrument-kafka-clients/

**Review by svrnm — APPROVED** [review]

**Comment by svrnm** [comment]
@ppatierno minor thing, see above, put nothing that stops us from mergin.

@open-telemetry/blog-approvers: please review & approve, thanks!

File: content/en/blog/2022/instrument-kafka-clients/index.md
Comment (ppatierno) [reply content/en/blog/2022/instrument-kafka-clients/index.md]: Makes sense. Done!

**Review by cartermp — APPROVED** [review]

**Review by austinlparker — APPROVED** [review]
Lgtm

**Pull request merged**
